sig-scalability/blogs/k8s-services-scalability-issues.md

```diff
@@ -0,0 +1,130 @@
+# Known Scalability Issues with Kubernetes Services
+
+_by Shyam JVS, Google Inc (with inputs from Brian Grant & Wojciech Tyczysnki)_
+
+**June 2018**
+
+This document serves as a catalog of issues we've known/discovered with kubernetes services as of today, focusing on their scalability/performance. The purpose of the document is to make the information common knowledge for the community, so we can work together towards improving it. Listing them below in no particular order.
+
+
+## Performance of iptables when large no. of service ports exist
+
+#### Issue
+
+Iptables can be slow in the presence of a large no. of services. As the no. of service ports increases, the KUBE-SERVICES chain gets longer and since it’s also evaluated very frequently that can bring down the performance. Also, it seems that manipulating a chain with lots of rules (e.g. KUBE-SERVICES) takes noticeably longer than others.
+
+#### Relevant issues/links
+
+- [#48938](https://github.com/kubernetes/kubernetes/issues/48938)
+
+#### Possible Solution(s)
+
+- Moving from iptables to IPVS may help here if that works stably. We have the IPVS alternative [implemented](https://github.com/kubernetes/kubernetes/pull/46580), but it still hasn’t gone to GA. Also we need to measure/confirm performance benefits from it.
+
+
+## Kube-proxy slow in updating iptables when large no. of backends exist
+
+#### Issue
+
+When we’re running services worth a total of 120k endpoints (backends) on our 5k-node scalability tests, we’re noticing that kube-proxy is timing out while trying to do iptables-restore due to failing to acquire lock over iptables.
+
+#### Relevant issues/links
+
+- [#48107](https://github.com/kubernetes/kubernetes/issues/48107)
+
+#### Possible Solution(s)
+
+- Give more time for kube-proxy to update iptables by increasing its batching interval for large clusters
```

Comment:
Batching interval is unrelated here - there will never be two executions of those in-flight (if there are, that's a bug, but looking into the code it shouldn't be the case).
So that doesn't seem to be an option.

What we should do is to verify where is this timeout coming from.
E.g. can you grab the contents and try running the command yourself? I don't know where this timeout is really coming from.